### PR TITLE
Add an option to preserve timestamps of the downloaded files (RhBug:1688537)

### PR DESCRIPTION
**Pull request by m-blaha**
If LRO_PRESERVETIME is set to True, librepo will try to keep the timestamps
of downloaded files in sync with that on the remote side.

https://bugzilla.redhat.com/show_bug.cgi?id=1688537

### CODE REVIEW

File: librepo/downloader.c [hunk 1]
```diff
@@ -18,7 +18,8 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
-#define _XOPEN_SOURCE   500 // Because of fdopen() and ftruncate()
+#define _BSD_SOURCE
+#define _DEFAULT_SOURCE
```
Comment (Conan-Kudo) [inline librepo/downloader.c]: Why did you drop `_XOPEN_SOURCE 500`?

Comment (m-blaha) [reply librepo/downloader.c]: I need futimes() to change file timestamps and this requires _DEFAULT_SOURCE (for glibc > 2.19) or _BSD_SOURCE (for glibc <= 2.19, which is the case for RHEL7).
Having _XOPEN_SOURCE explicitly defined means, that _DEFAULT_SOURCE is not defined.

Comment (Conan-Kudo) [reply librepo/downloader.c]: How does this affect non-glibc environments?

Comment (m-blaha) [reply librepo/downloader.c]: I'm not sure. But I can return _XOPEN_SOURCE   500 back

Comment (Conan-Kudo) [reply librepo/downloader.c]: It shouldn't hurt anything to put it back, and I'm wary of accidentally breaking non-glibc builds (there are a few out there, including Yocto...).

**Comment by Conan-Kudo** [comment]
@rh-atomic-bot r+

**Comment by jamescassell** [comment]
Seems like it would be better to enable this option by default or always use this Behavior with an out an option to closer match python URL grabber from yum 3

**Pull request merged**